zstacklib/zstacklib/utils/ssh_validation.py

```
@@ -0,0 +1,156 @@
+
+def validate_ssh_port(port):
+    if port is None:
+        raise SSHValidationError("SSH port number cannot be empty")
+    if isinstance(port, bool):
+        raise SSHValidationError("SSH port number must be an integer, not bool")
+    try:
+        port_int = int(port)
+    except (ValueError, TypeError):
+        raise SSHValidationError(
+            "SSH port number must be an integer, current value: {0} (type: {1})".format(port, type(port).__name__)
+        )
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)